src/bluehawk/actions/run.ts

```diff
@@ -0,0 +1,248 @@
+            waitForListeners: true,
+          });
+
+          break;
+
+        case "check":
+          await check({
+            paths: [source],
+            ignore: ignore,
+            json: json,
+            reporter,
+          });
+
+          break;
+
+        case "copy":
+          await copy({
+            reporter,
+            output: destination,
+            rootPath: source,
+            waitForListeners: true,
+          });
+
+          break;
+
+        default:
+          console.error(new Error("No Bluehawk command found!"));
+
+          break;
+      }
+    }
+
+    await processSubconfigFiles();
+  } else {
+    console.error("No Bluehawk config commands found.");
+  }
+
+  // Look for and process config files in child directories
+  async function processSubconfigFiles() {
+    const ignores: string[] = [".git"];
+
+    try {
+      const gitignore = await System.fs.readFile(
+        Path.join(currentWorkingDirectory, ".gitignore"),
+        "utf8"
+      );
+      const gitignores = gitignore
+        .split(/\r\n|\r|\n/)
+        .filter((line) => !(line.startsWith("#") || line.trim().length == 0));
+
+      ignores.push(...gitignores);
+    } catch {
+      const gitignores = [
+        "android",
+        "ios",
+        ".*",
+        "*.lock",
+        "gemfile",
+        "*.config.*",
+        "*.json",
+        "*.cjs",
+        "*.md",
+      ];
+
+      ignores.push(...gitignores);
+    }
+
+    const filePaths = await loadProjectPaths({
+      rootPath: currentWorkingDirectory,
+      ignore: ignores,
+    });
+
+    // Look for file paths that:
+    // - include "bluhawk.config.yaml"
+    // - do not include the current working directory + "bluhawk.config.yaml"
+    // - the last bit of the file path equals "bluhawk.config.yaml"
+    const subConfigFilePaths = filePaths.filter(
+      (filePath) =>
+        filePath.includes(configFileName) &&
+        filePath != Path.join(currentWorkingDirectory, configFileName) &&
+        filePath.split("/").pop() == configFileName
+    );
+
+    // Process each subconfig file
+    for (let index = 0; index < subConfigFilePaths.length; index++) {
+      const subConfig = await parseConfigFile(subConfigFilePaths[index]);
+      const subConfigPath = subConfigFilePaths[index].substring(
+        0,
+        subConfigFilePaths[index].lastIndexOf(configFileName)
+      );
+
+      reporter.onFileParsed({
+        inputPath: subConfigFilePaths[index],
+        isConfig: true,
+      });
+
+      // Run config commands
+      for (let index = 0; index < subConfig.commands.length; index++) {
+        const { command, source, destination, ignore, state, format, json } =
+          subConfig.commands[index];
+        const sourceFilePath = source
+          ? Path.join(subConfigPath, source)
+          : subConfigPath;
+
+        reporter.onActionProcessed({
+          inputPath: sourceFilePath,
+          name: command,
+        });
+
+        switch (command) {
```

Comment:
We ought to refactor this since it's using more or less the same code to do the same thing but in two places.